File: files/2025-08-31/CSEC 31082025.R
library(tidyverse)
library(readxl)

path = "files/2025-08-31/Challenge 53.xlsx"
input = read_excel(path, range = "B2:E8")
test  = read_excel(path, range = "G2:G4") %>% pull()

result = input %>%
  filter(if_all(-Customer, is.na)) %>%
  pull(Customer)

all.equal(result, test)

File: files/2025-08-31/CSEC 31082025.py
import pandas as pd

path = "files/2025-08-31/Challenge 53.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="G", skiprows=1, nrows=2).squeeze().tolist()

result = input[input.drop('Customer', axis=1).isna().all(axis=1)]['Customer'].tolist()

print(test == result) # True